airflow/models.py

```diff
@@ -2031,20 +2008,32 @@ def date_range(self, start_date, num=None, end_date=datetime.now()):
             start_date=start_date, end_date=end_date,
             num=num, delta=self._schedule_interval)
 
-    def following_schedule(self, dttm):
+    def directly_following_schedule(self, dttm):
         if isinstance(self._schedule_interval, six.string_types):
             cron = croniter(self._schedule_interval, dttm)
             return cron.get_next(datetime)
         elif isinstance(self._schedule_interval, timedelta):
             return dttm + self._schedule_interval
 
-    def previous_schedule(self, dttm):
+    def following_schedule(self, dttm, n=1):
+        following = dttm
+        for i in range(0, n):
+            following = self.directly_following_schedule(following)
+        return following
+
+    def directly_previous_schedule(self, dttm):
         if isinstance(self._schedule_interval, six.string_types):
             cron = croniter(self._schedule_interval, dttm)
             return cron.get_prev(datetime)
         elif isinstance(self._schedule_interval, timedelta):
             return dttm - self._schedule_interval
 
+    def previous_schedule(self, dttm, n=1):
+        previous = dttm
```

Comment:
Same remark for the `n` + how about a recursive approach to avoid the mutating for-loop, something like: 
```
    def previous_schedule(self, dttm, n=1):

        prev_sched = self.directly_previous_schedule(dttm)
        
        if n = 1:
            return prev_sched

        else:            
            previous_schedule(self, prev_sched, n-1):

```

(same note for the `following` version)